src/providers/swift.rs

```diff
@@ -0,0 +1,177 @@
+
+        if app.includes_file("Package.resolved") {
+            install_phase.add_file_dependency("Package.resolved".to_string());
+        }
+
+        Ok(Some(install_phase))
+    }
+
+    fn build(&self, app: &App, _env: &Environment) -> Result<Option<BuildPhase>> {
+        let name = SwiftProvider::get_executable_name(app)?;
+        let build_cmd = format!(
+            "\
+        CC=clang++ \
+        swift build -c release --static-swift-stdlib && \
+        cp ./.build/release/{name} ./{name} && \
+        rm -rf ./.build
+        "
+        )
+        .trim()
+        .to_string();
+
+        Ok(Some(BuildPhase::new(build_cmd)))
+    }
+
+    fn start(&self, app: &App, _env: &Environment) -> Result<Option<StartPhase>> {
+        let name = SwiftProvider::get_executable_name(app)?;
+
+        Ok(Some(StartPhase::new(format!("./{}", name))))
+    }
+}
+
+impl SwiftProvider {
+    fn get_swift_version(app: &App) -> Result<String> {
+        if app.includes_file(".swift-version") {
+            let contents = app.read_file(".swift-version")?;
+            let version = contents.split('\n').collect::<Vec<_>>();
+
+            match version.first() {
+                Some(v) => Ok(v.trim().to_string()),
+                None => bail!("Your .swift-version file is empty"),
+            }
+        } else if app.includes_file("Package.swift") {
+            let contents = app.read_file("Package.swift")?;
+            let version = contents
+                .split('\n')
+                .filter(|&l| l.contains("swift-tools-version:"))
+                .map(|l| {
+                    l.replace("swift-tools-version:", "")
+                        .replace("//", "")
+                        .trim()
+                        .to_string()
+                })
+                .collect::<Vec<_>>()
+                .first()
+                .map(|s| s.to_owned());
+
+            if let Some(version) = version {
+                Ok(version)
+            } else {
+                Ok(DEFAULT_SWIFT_VERSION.to_string())
+            }
+        } else {
+            Ok(DEFAULT_SWIFT_VERSION.to_string())
+        }
+    }
+
+    fn get_executable_name(app: &App) -> Result<String> {
+        let raw_paths = app.find_files("Sources/**/main.swift")?;
+        let paths = raw_paths
+            .iter()
+            .filter(|&path| !path.to_string_lossy().contains(".build"))
+            .collect::<Vec<_>>();
+
+        let path = match paths.first() {
+            Some(path) => path.to_string_lossy().to_string(),
+            None => bail!("Your swift app doesn't have a main.swift file"),
+        };
+
+        let mut names = path.split('/').collect::<Vec<_>>();
+
+        // Safe to unwrap, because the path was filtered by the glob expression
+        let pos = names.iter().position(|&n| n == "Sources").unwrap();
+
+        names.drain(0..pos);
+
+        Ok(names[1].to_string())
+    }
+}
+
+fn version_number_to_rev(version: &str) -> Result<Option<String>> {
```

Comment:
This function shouldn't return a `Result`, as there is nowhere in the code where it can return an `Err`.